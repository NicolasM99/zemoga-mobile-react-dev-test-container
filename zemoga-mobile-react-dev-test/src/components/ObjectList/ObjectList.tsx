import React, { FC } from 'react';
import { Text, View } from 'react-native';
import { IObjectList } from './@types/objectList';
import { objectToList } from './util/objectToList';

const ObjectList: FC<IObjectList> = ({ objectData }: IObjectList) => {
  const objectList = objectToList(objectData);
  const printSpaces = (level: number = 0) => {
    let spaces = '';
    if (!level) return '';
    for (let i = 0; i < level; i++) {
      spaces += '  ';
    }
    return spaces;
  };
  return (
    <View>
      {objectList.map(({ key, value, level }: any) => (
        <Text style={{ marginTop: !value ? 10 : 4 }}>
          <Text style={{ fontWeight: 'bold' }}>
            {printSpaces(level)}
            {key}:{' '}
          </Text>
          {value}
        </Text>
      ))}
    </View>
  );
};

export default ObjectList;
