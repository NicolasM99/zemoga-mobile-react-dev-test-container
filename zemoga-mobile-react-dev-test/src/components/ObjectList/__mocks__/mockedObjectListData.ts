export const mockedObjectList = {
  test: 'Test attribute',
  nested0: {
    nestedlvl0attribute: 'Nested level 0 attribute',
    nested1: {
      nestedlvl1attribute: 'Nested level 1 attribute'
    }
  }
};
