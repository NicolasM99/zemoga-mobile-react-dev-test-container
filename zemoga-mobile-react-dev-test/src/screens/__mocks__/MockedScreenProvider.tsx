import React from 'react';

import { NavigationContainer } from '@react-navigation/native';

import { PostsListProvider } from 'src/components/PostsList/context/PostsListContext';
import { InternetStatusProvider } from 'src/context/InternetStatusContext';

interface MockedScreenProviderProps {
  children: React.ReactNode;
}

const MockedScreenProvider: React.FC<MockedScreenProviderProps> = ({
  children
}: MockedScreenProviderProps) => (
  <NavigationContainer>
    <InternetStatusProvider>
      <PostsListProvider>{children}</PostsListProvider>
    </InternetStatusProvider>
  </NavigationContainer>
);

export default MockedScreenProvider;
