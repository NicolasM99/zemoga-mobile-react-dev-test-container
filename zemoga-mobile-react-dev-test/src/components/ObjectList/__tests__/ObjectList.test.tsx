/* eslint-disable @typescript-eslint/naming-convention */
import { render, screen } from '@testing-library/react-native';

import { mockedObjectList } from '../__mocks__/mockedObjectListData';
import ObjectList from '../ObjectList';

describe('<ObjectList /> unit tests', () => {
  it('Should mount correctly the component', () => {
    render(<ObjectList objectData={{}} />);
    expect(screen).toMatchSnapshot();
  });

  it('Should display correctly the objectData', async () => {
    render(<ObjectList objectData={mockedObjectList} />);
    const test_0 = await screen.findByTestId('test_0');
    const nested0_0 = await screen.findByTestId('nested0_0');
    const nestedlvl0attribute_1 = await screen.findByTestId('nestedlvl0attribute_1');
    const nested1_1 = await screen.findByTestId('nested1_1');
    const nestedlvl1attribute_2 = await screen.findByTestId('nestedlvl1attribute_2');
    expect(
      `${test_0.props.children[0].props.children.join('')}${test_0.props.children[1]}`
    ).toEqual('Test: Test attribute');
    expect(`${nested0_0.props.children[0].props.children.join('')}`).toEqual('Nested0: ');
    expect(`${nested1_1.props.children[0].props.children.join('')}`).toEqual('  Nested1: ');
    expect(
      `${nestedlvl0attribute_1.props.children[0].props.children.join('')}${
        nestedlvl0attribute_1.props.children[1]
      }`
    ).toEqual('  Nestedlvl0attribute: Nested level 0 attribute');
    expect(
      `${nestedlvl1attribute_2.props.children[0].props.children.join('')}${
        nestedlvl1attribute_2.props.children[1]
      }`
    ).toEqual('    Nestedlvl1attribute: Nested level 1 attribute');
  });
});
