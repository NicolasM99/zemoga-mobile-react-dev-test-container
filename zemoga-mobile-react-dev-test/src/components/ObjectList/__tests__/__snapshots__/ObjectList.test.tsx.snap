// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ObjectList /> unit tests Should mount correctly the component 1`] = `<View />`;
